File: leetcode/lc-us/cpp/57.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>> &intervals, vector<int> &newInterval) {
        vector<vector<int>> res;
        bool used = false;
        int left = newInterval[0], right = newInterval[1];
        for (vector<int> &interval: intervals) {
            if (interval[0] > right) {
                if (!used) {
                    res.push_back({left, right});
                    used = true;
                }
                res.push_back(interval);
            } else if (interval[1] < left) res.push_back(interval);
            else {
                left = min(left, interval[0]);
                right = max(right, interval[1]);
            }
        }
        if (!used) res.push_back({left, right});
        return res;
    }
};

File: leetcode/lc-us/py/57.py
from typing import List


class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        left, right = newInterval
        placed = False
        res = list()
        for interval in intervals:
            if interval[0] > right:
                if not placed:
                    res.append([left, right])
                    placed = True
                res.append(interval)
            elif interval[1] < left:
                res.append(interval)
            else:
                left = min(left, interval[0])
                right = max(right, interval[1])
        if not placed:
            res.append([left, right])
        return res
